rfd/0xxx-kube-mfa-sessions.md

```diff
@@ -0,0 +1,142 @@
+---
+    authors: Anton Miniailo (anton@goteleport.com)
+    state: draft
+---
+
+# RFD 0xxx - Kubernetes MFA sessions
+
+## Required Approvers
+
+- Engineering: `@r0mant` && `@tigrato`
+- Product: `@klizhentas` || `@xinding33`
+- Security: `@reedloden`
+
+
+## What
+Use local proxy for Kubernetes access to extend lifetime of Kubernetes MFA sessions, so that users don't have to use MFA device for every command they run when per-session-MFA is enabled.
+
+## Why
+Kubernetes commands currently are treated as separate MFA sessions, so each time user runs a kubectl command they have to do a MFA check. This creates a burdensome user experience.
+By using local Kubernetes proxy we can extend MFA session to be limited not by running single command but by lifetime of the local proxy. We should also remove the 1 minute restriction 
+on TTL of Kubernetes certificates for local proxy to improve overall UX.
+
+## Details
+
+This proposal builds heavily upon [RFD 90 "Database MFA Sessions"](https://github.com/gravitational/teleport/blob/master/rfd/0090-db-mfa-sessions.md). 
+It is an extension of the principles established in RFD 90 and it follows the same structure and
+approach since we're extending same idea to provide very similar functionality, but for Kubernetes access.
+
+Local proxy for Kubernetes access will handle issuing MFA certificates and will keep them in memory while the proxy process is running. 
+As with DB MFA sessions, certificates TTL will be limited to the `max_session_ttl` time instead of 1 minute.
+Local proxy will also reissue certificates and ask for MFA check/credentials if required.
+
+Unlike DB MFA sessions, Kubernetes local proxy will only have one mode of operation (DB access has tunnelled and not tunnelled mode). 
+Since the pattern of usage for Kubernetes is mostly short-lived commands, analogue of tunneled mode for the local proxy makes the most sense.
```

Comment:
Didn't @greedy52 already introduced the non-tunneled mode? 

If so, we should differentiate the behavior and by default disable the MFA caching.
Kube local proxy is still required when running behind an L7 load balancer and we don't want to enable MFA caching by default when running behind L7 load balancers.

If users want MFA caching, they must explicitly enable it.
This changes nothing in terms of code, it's just an extra flag.

It probably requires some UX to inform kubectl when a certificate must be reissued

Reply:
> analogue of tunneled mode for the local proxy makes the most sense

ya. The definition of tunnelled vs non-tunnelled for DB is probably different than Kube. Tunnelled for DB means database client sends data to local proxy in plain database protocol and local proxy will take care of the mTLS part, where non-tunnelled means database client sends encrypted database with Teleport-issued cert.

I would say `tsh proxy kube` is closer to tunnelled mode since kube clients are not using Teleport-issued cert against local proxy, and local proxy is handling the Teleport-issued cert.

Reply:
Yep, as @greedy52 said, what was currently added is already I would say tunneled mode. I think having only this mode makes sense for Kube.